### PR TITLE
feat(api): finish update-my-email endpoint

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

When this endpoint was first migrated, we had not implemented the email service. That now exists, so I could finish endpoint.

Next job is to create the /confirm-email endpoint, without which this doesn't do a lot!

### CODE REVIEW

File: api/src/routes/settings.test.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
 import {
   devLogin,
   setupServer,
   superRequest,
-  createSuperRequest
+  createSuperRequest,
+  defaultUserId
 } from '../../jest.utils';
 import { createUserInput } from '../utils/create-user';
```
Comment (ShaunSHamilton) [inline api/src/routes/settings.test.ts:10]:
```suggestion
import { API_LOCATION } from '../utils/env';
```

File: api/src/routes/settings.test.ts [hunk 2]
```diff
@@ -317,7 +330,60 @@ Please wait 5 minutes to resend an authentication link.`
...
+            // interested in is the id.
+            Promise.resolve({
+              id: '123'
+            })
+          );
+        await superPut('/update-my-email').send({
+          email: unusedEmailOne
+        });
+
+        const expectedLink = `http://localhost:3000/confirm-email?email=${Buffer.from(unusedEmailOne).toString('base64')}&token=123&emailChange=true`;
```
Comment (ShaunSHamilton) [inline api/src/routes/settings.test.ts:347]:
```suggestion
        const expectedLink = `${API_LOCATION}/confirm-email?email=${Buffer.from(unusedEmailOne).toString('base64')}&token=123&emailChange=true`;
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
I did not want to hold this up, but, at the same time, knew I would not remember to fix this until it annoyed me again.

File: api/src/routes/settings.test.ts (hunk 2 above)
Comment (ojeytonwilliams) [reply api/src/routes/settings.test.ts]: Thanks. I shouldn't have been lazy.

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

**Review by Sembauke — APPROVED** [review]

**Pull request merged**